src/storage/txn/commands/compare_and_set.rs

```diff
@@ -0,0 +1,135 @@
+// Copyright 2021 TiKV Project Authors. Licensed under Apache-2.0.
+
+use crate::storage::kv::{Modify, WriteData};
+use crate::storage::lock_manager::LockManager;
+use crate::storage::txn::commands::{
+    Command, CommandExt, ResponsePolicy, TypedCommand, WriteCommand, WriteContext, WriteResult,
+};
+use crate::storage::txn::Result;
+use crate::storage::{ProcessResult, Snapshot};
+use engine_traits::CfName;
+use txn_types::{Key, Value};
+
+command! {
+    /// CompareAndSet to check if whether the previous value of the key equals to the given value.
+    /// If them equal, write the new value, otherwise return the previous value.
```

Comment:
```suggestion
    /// CompareAndSet to check whether the previous value of the key equals to the given value.
    /// If they are equal, write the new value, otherwise return the previous value.
```